lib/ex_aws/sqs/parsers.ex

```diff
@@ -0,0 +1,204 @@
+                            value: ~x"./Value/text()"s |> SweetXml.transform_by(&try_cast_to_number/1)
+                          ]
+                        ])
+      fixed_attributes = case get_in(parsed_body, [:message, :attributes]) do
+                           nil ->
+                             parsed_body
+
+                             v when is_list(v) ->
+                             update_in(parsed_body, [:message, :attributes], &attribute_list_to_map/1)
+                         end
+
+      {:ok, Map.put(resp, :body, fixed_attributes)}
+    end
+
+    def parse({:ok, resp=%{body: xml}}, :send_message) do
+      parsed_body = xml
+      |> SweetXml.xpath(~x"//SendMessageResponse",
+                        request_id: request_id_xpath(),
+                        message_id: ~x"./SendMessageResult/MessageId/text()"s,
+                        md5_of_message_body: ~x"./SendMessageResult/MD5OfMessageBody/text()"s,
+                        md5_of_message_attributes: ~x"./SendMessageResult/MD5OfMessageAttributes/text()"s,
+      )
+
+      {:ok, Map.put(resp, :body, parsed_body)}
+    end
+
+    def parse({:ok, resp=%{body: xml}}, :send_message_batch) do
+      parsed_body = xml
+      |> SweetXml.xpath(~x"//SendMessageBatchResponse",
+                        request_id: request_id_xpath(),
+                        successes: [
+                          ~x"./SendMessageBatchResult/SendMessageBatchResultEntry"l,
+                          id: ~x"./Id/text()"s,
+                          message_id: ~x"./MessageId/text()"s,
+                          md5_of_message_body: ~x"./MD5OfMessageBody/text()"s,
+                          md5_of_message_attributes: ~x"./MD5OfMessageAttributes/text()"s
+                        ])
+      {:ok, Map.put(resp, :body, parsed_body)}
+
+    end
+
+    def parse(resp={:ok, _}, :set_queue_attributes) do
+      parse_request_id(resp, ~x"//SetQueueAttributesResponse")
+    end
+
+    def parse({:error, {type, http_status_code, xml}}, _) do
+      parsed_body = xml
+      |> SweetXml.xpath(~x"//ErrorResponse",
+                        request_id: ~x"./RequestId/text()"s,
+                        type: ~x"./Error/Type/text()"s,
+                        code: ~x"./Error/Code/text()"s,
+                        message: ~x"./Error/Message/text()"s,
+                        detail: ~x"./Error/Detail/text()"s)
+
+      {:error, {type, http_status_code, parsed_body}}
+    end
+
+    def parse(val, _), do: val
+
+    def parse_request_id({:ok, resp=%{body: xml}}, parent_xpath) do
+      parsed_body = xml
+      |> SweetXml.xpath(parent_xpath,
+                        request_id: request_id_xpath())
+
+      {:ok, Map.put(resp, :body, parsed_body)}
+    end
+
+    def attribute_list_to_map(list_of_maps) do
+      list_of_maps
+      |> Enum.reduce(%{}, fn(%{name: name, value: val}, acc) ->
+        attribute_name = name
+        |> Macro.underscore
+        |> String.to_atom
```

Comment:
The other `String.to_atom` call is fine cause all the inputs are basically hardcoded.

Aren't attribute lists basically arbitrary input?